beacon_node/http_api/src/lib.rs

```diff
@@ -3410,11 +3427,14 @@ pub fn serve<T: BeaconChainTypes>(
         .and(network_tx_filter.clone())
         .and(log_filter.clone())
         .then(
-            |task_spawner: TaskSpawner<T::EthSpec>,
+            |not_synced_filter: Result<(), Rejection>,
+             task_spawner: TaskSpawner<T::EthSpec>,
              chain: Arc<BeaconChain<T>>,
              aggregates: Vec<SignedAggregateAndProof<T::EthSpec>>,
              network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>, log: Logger| {
                 task_spawner.blocking_json_task(Priority::P0, move || {
+                    if let Err(e) = not_synced_filter { return Err(e) }
```

Comment:
`cargo fmt` doesn't like this syntax

I wonder if we could get away with:

```rust
not_synced_filter?;
```

Clippy might think that's odd, but I think it's perfectly normal :grin: